src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -70,11 +81,11 @@ public async Task StartAsync(CancellationToken cancellationToken)
                 throw new InvalidOperationException("The listener has already started.");
             }
 
-            this.InitializeHost();
+            this.InitializeBuilder();
 
             try
             {
-                await RegisterObserverFactoryAsync();
+                await this.StartProcessorAsync();
```

Comment:
If this throws an exception, are we able to try again with the same host? I'm thinking about the cases where we re-try listeners if they fail to start up.